pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -100,54 +86,230 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		// error reading the object - requeue the request
 		return reconcile.Result{}, err
 	}
 
-	// Check if this Pod already exists
-	found := &corev1.Pod{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
-		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		err = r.client.Create(context.TODO(), pod)
-		if err != nil {
-			return reconcile.Result{}, err
+	// get permission type, empty at first
+	permission := &managedv1alpha1.Permission{}
+
+	// get clusterRoleName from permission type
+	clusterRoleName := permission.ClusterRoleName
+
+	// if crClusterRoleNameList returns true with values in there
+	allClusterRoleNameInCluster, crClusterRoleNameList := checkClusterRoleName(instance, clusterRoleList)
+	if allClusterRoleNameInCluster == false {
+		for _, crClusterRoleName := range crClusterRoleNameList {
+			// get the conditions array
+			groupPermissionConditions := instance.Status.Conditions
+
+			// make new condition
+			newCondition := managedv1alpha1.Condition{
+				LastTransitionTime: metav1.Now(),
+				Message:            "ClusterRole for clusterPermission does not exist",
+				ClusterRoleName:    crClusterRoleName,
+				Status:             true,
+				State:              "Failed",
+			}
+
+			// append new condition to conditions array
+			groupPermissionConditions = append(groupPermissionConditions, newCondition)
+
+			// update GroupPermission object with new conditions
+			instance.Status.Conditions = groupPermissionConditions
+
+			err = r.client.Status().Update(context.TODO(), instance)
+			if err != nil {
+				reqLogger.Error(err, "Failed to update condition.")
+				return reconcile.Result{}, err
+			}
 		}
+	}
 
-		// Pod created successfully - don't requeue
-		return reconcile.Result{}, nil
-	} else if err != nil {
+	// get a list of clusterRoleBinding from k8s cluster list
+	clusterRoleBindingList := &v1.ClusterRoleBindingList{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, clusterRoleBindingList)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
 		return reconcile.Result{}, err
 	}
 
-	// Pod already exists - don't requeue
-	reqLogger.Info("Skip reconcile: Pod already exists", "Pod.Namespace", found.Namespace, "Pod.Name", found.Name)
+	// build a clusterRoleBindingNameList which consists of clusterRoleName-groupName
+	crClusterRoleBindingNameList := buildClusterRoleBindingName(instance)
+
+	// check ClusterRoleBindingName
+	allClusterRoleBindingNameInCluster, crClusterRoleBindingNameList := checkClusterRoleBindingName(crClusterRoleBindingNameList, clusterRoleBindingList)
+	if allClusterRoleBindingNameInCluster == false {
+		newCRB := newClusterRoleBinding(instance.Spec.GroupName, permission)
```

Comment:
Put in a for loop to iterate through needed clusterrolebindings. Make function definition take a single string.